modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/PlayerShelf.kt

```diff
@@ -0,0 +1,535 @@
+    onClick: () -> Unit,
+) {
+    IconButton(onClick = onClick) {
+        Icon(
+            painterResource(id = R.drawable.ic_share_android_32),
+            contentDescription = stringResource(LR.string.share_podcast),
+            tint = iconColors.normalColor,
+        )
+    }
+}
+
+@Composable
+fun PodcastButton(
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    IconButton(onClick = onClick) {
+        Icon(
+            painterResource(id = IR.drawable.ic_goto_32),
+            contentDescription = stringResource(LR.string.go_to_podcast),
+            tint = iconColors.normalColor,
+        )
+    }
+}
+
+@Composable
+fun CastButton(
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    AndroidView(
+        factory = { context ->
+            MediaRouteButton(context).apply {
+                CastButtonFactory.setUpMediaRouteButton(context, this)
+                visibility = View.VISIBLE
+                updateColor(iconColors.normalColor.toArgb())
+                setOnClickListener { onClick() }
+                CastButtonFactory.setUpMediaRouteButton(context, this)
+            }
+        },
+    )
+}
+
+@Composable
+fun PlayedButton(
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    IconButton(onClick = onClick) {
+        Icon(
+            painterResource(id = R.drawable.ic_tick_circle_ol_32),
+            contentDescription = stringResource(LR.string.mark_as_played),
+            tint = iconColors.normalColor,
+        )
+    }
+}
+
+@Composable
+fun ArchiveButton(
+    isUserEpisode: Boolean,
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    val archiveResource = (if (isUserEpisode) R.drawable.ic_delete_32 else R.drawable.ic_archive_32)
+    IconButton(onClick = onClick) {
+        Icon(
+            painterResource(id = archiveResource),
+            contentDescription = stringResource(LR.string.archive_episode),
+            tint = iconColors.normalColor,
+        )
+    }
+}
+
+@Composable
+fun BookmarkButton(
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    IconButton(onClick = onClick) {
+        Icon(
+            painterResource(id = IR.drawable.ic_bookmark),
+            contentDescription = stringResource(LR.string.add_bookmark),
+            tint = iconColors.normalColor,
+        )
+    }
+}
+
+@Composable
+fun TranscriptButton(
+    isTranscriptAvailable: Boolean,
+    iconColors: PlayerShelfIconColors,
+    onClick: (Boolean) -> Unit,
+) {
+    val alpha = if (isTranscriptAvailable) 1f else 0.4f
+    IconButton(onClick = { onClick(isTranscriptAvailable) }) {
+        Icon(
+            painterResource(id = IR.drawable.ic_transcript_24),
+            contentDescription = stringResource(LR.string.transcript),
+            tint = iconColors.normalColor,
+            modifier = Modifier.alpha(alpha),
+        )
+    }
+}
+
+@Composable
+fun DownloadButton(
+    isPodcast: Boolean,
```

Comment:
Could you name it `isPodcastEpisode`? When I was reviewing it at one point I had to pause here to understand that it is related to episodes.